### PR TITLE
added multiple tests, new logic on comments model, refactored the log…

### PR DESCRIPTION
**Pull request by Antonio-Bogalho**
…ic for the other tests, new controller to add comments by id, added new error catcher with a new error code in app.js

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -149,4 +149,79 @@ describe("/api/articles", () => {
...
+        .send({
+          username: "butter_bridge",
+          body: "body-test-123",
+        })
+        .set("Accept", "application/json")
+        .expect(201)
+        .then(({ body }) => {
+          const { comments } = body;
+          expect(Array.isArray(comments)).toBe(true);
+          expect(comments).toHaveLength(1);
```
Comment (nc-mark-t) [inline __tests__/app.test.js:165]: You should respond with the comment on a key of `comment` and not in an array

File: __tests__/app.test.js [hunk 2]
```diff
@@ -149,4 +149,79 @@ describe("/api/articles", () => {
...
+      .send({
+        username: "not-a-username",
+        body: "body-test-123"
+      })
+      .set("Accept", "application/json")
+      .expect(404)
+      .then(({body}) => {
+        expect(body.msg).toBe('Not found');
+      });
+  });
```
Comment (nc-mark-t) [inline __tests__/app.test.js:226]: You might want to check that additional keys are ignored

File: controllers/comments.controller.js [hunk 3]
```diff
@@ -9,4 +9,20 @@ exports.getCommentsByArticleId = (req, res, next) => {
...
+    const articleId = req.params.article_id;
+    const { body, username } = req.body;
+
+    return getComments(articleId)
+        .then(() => {
+
+            return addComments(body, username, articleId);
+        })
+        .then((newComment) => {
+            res.status(201).send({ comments: [newComment] });
```
Comment (nc-mark-t) [inline controllers/comments.controller.js:23]: Here, we don't need the response to be in an array, a json object is fine, also `comments` suggests that there are multiple

**Review by nc-mark-t — COMMENTED** [review]
See my comments.

**Comment by Antonio-Bogalho** [comment]
thank you for the feedback, ill make sure to have it sorted for the next ticket 👍

**Pull request merged**